### PR TITLE
Feature 138 docx report export

### LINKED ISSUE #138: DOCX document export
As a user I need to be able to export a docx document out of the survey

### PR DESCRIPTION
**Pull request by Daniilmass**
Closes #138

### CODE REVIEW

File: src/components/docx-export/docx-export-utils.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    ownerAndBuildingInfoPage,
+    reusableMaterialsPage,
+    wasteMaterialsPage,
+    hazardousMaterialsPage
+  ];
+
+  /**
+   * Unpack page functions
+   * 
+   * @param doc document
```
Comment (EVP1989) [inline src/components/docx-export/docx-export-utils.tsx:26]: Add other parameters to docs

File: src/components/docx-export/docx-export-utils.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    for (const pageFunction of pageFunctions) {
+      // eslint-disable-next-line no-await-in-loop
+      await pageFunction(doc, survey, surveySummary, localization);
+    }
+  };
+
+  /**
+   * Downlaod docx file to client's device
+   * 
+   * @param doc document
```
Comment (EVP1989) [inline src/components/docx-export/docx-export-utils.tsx:40]: same here

File: src/components/docx-export/docx-export-utils.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    const tempLink = document.createElement("a");
+    const surveyName = survey.building?.propertyName;
+
+    tempLink.href = csvURL;
+    tempLink.setAttribute("download", `${surveyName ?? strings.docx.metadata.unnamed}.docx`);
+    tempLink.click();
+  };
+
+  /**
+   * Generate docx file
```
Comment (EVP1989) [inline src/components/docx-export/docx-export-utils.tsx:54]: Same here

File: src/components/docx-export/helpers/image-utils.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+    return pictures;
+  };
+
+  /**
+   * Get attachments collection
+   * 
+   * @param doc docx document
+   * @param attachments survey summary attachments
+   * @returns
```
Comment (EVP1989) [inline src/components/docx-export/helpers/image-utils.tsx:100]: Add return value

File: src/components/docx-export/pages/demolition-information-page.tsx [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+        borders: DocumentTableStyles.noBorders
+      })
+    ]
+  });
+
+  // Assemble the table rows
+  surveyInfoTableRows.push(propertyNameRow);
+  surveyInfoTableRows.push(scopeRow);
+  surveyInfoTableRows.push(startDateRow);
+  surveyInfoTableRows.push(endDateRow);
```
[the remark this answers is not in the document]
Comment (EVP1989) [reply src/components/docx-export/pages/demolition-information-page.tsx:169]: I think it should work also like this.

File: src/components/docx-export/pages/hazardous-materials-page.tsx [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+                text: hazardousWaste.description,
+                style: "Normal"
+              })
+            ],
+            borders: DocumentTableStyles.noBorders
+          })
+        ]
+      });
+
+      hazardousMaterialsTableRows.push(hazardousMaterialTableRow);
```
Comment (EVP1989) [inline src/components/docx-export/pages/hazardous-materials-page.tsx:198]: Check syntax from previous file

File: src/components/docx-export/pages/hazardous-materials-page.tsx [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+import { Survey } from "generated/client";
+import { SurveySummary } from "types";
+import PageLayout from "../helpers/page-layout";
+import strings from "localization/strings";
+import LocalizationUtils from "utils/localization-utils";
+import DocumentTableStyles from "../styles/document-table-styles";
+
+/**
+ * Hazardous materials Page
+ *
```
Comment (EVP1989) [inline src/components/docx-export/pages/hazardous-materials-page.tsx:11]: Add parameters to docs

File: src/components/docx-export/pages/owner-and-building-info-page.tsx [hunk 8]
```diff
@@ -0,0 +1,650 @@
...
+            ]
+          })
+        ],
+        borders: DocumentTableStyles.noBorders
+      })
+    ]
+  });
+
+  // Assemble the table rows
+  buildingInfoTableRows.push(buildingIDRow);
```
Comment (EVP1989) [inline src/components/docx-export/pages/owner-and-building-info-page.tsx:536]: Check syntax

File: src/components/docx-export/pages/reusable-materials-page.tsx [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+    pageChildren.push(dividerPar);
+
+    pageChildren.push(
+      new Paragraph({
+        children: reusableImages
+      })
+    );
+  });
+
+  // Here we push our table to the page children
```
Comment (EVP1989) [inline src/components/docx-export/pages/reusable-materials-page.tsx:233]: Do we need this commented code?

File: src/components/docx-export/pages/waste-materials-page.tsx [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+                text: waste.description,
+                style: "Normal"
+              })
+            ],
+            borders: DocumentTableStyles.noBorders
+          })
+        ]
+      });
+
+      wasteMaterialsTableRows.push(wasteMaterialTableRow);
```
Comment (EVP1989) [inline src/components/docx-export/pages/waste-materials-page.tsx:196]: Check syntax

**Review by EVP1989 — COMMENTED** [review]
Approved with comments and suggestions

File: src/localization/en.json [hunk 11]
```diff
@@ -499,11 +499,40 @@
...
+      "unnamed": "Unnamed"
+    },
+    "page1": {
+      "title": "Demolition information"
+    },
+    "page2": {
+      "title": "Building information",
+      "ownerContactPerson": "Omistajan yhteyshenkilö:",
+      "ownerAndAddress": "Rakennuksen omistaja ja osoite:",
+      "buildingInfo": "Purettavan rakennuksen tiedot:"
```
Comment (lauriLukkarinen) [inline src/localization/en.json:524]:
```suggestion
      "ownerContactPerson": "Owner's contact person:",
      "ownerAndAddress": "Building owner and address:",
      "buildingInfo": "Building info:"
```

**Review by lauriLukkarinen — CHANGES_REQUESTED** [review]

File: src/components/docx-export/docx-export-utils.tsx [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+  };
+
+  /**
+   * Downlaod docx file to client's device
+   * 
+   * @param doc document
+   */
+  const clientDocDownload = async (doc: Document, survey: SurveySummary) => {
+    const docBlob = await Packer.toBlob(doc);
+    const csvURL = window.URL.createObjectURL(docBlob);
```
[the remark this answers is not in the document]
Comment (Kaemppi) [reply src/components/docx-export/docx-export-utils.tsx:44]: I dont know. Was done by Daniil.

File: src/components/docx-export/pages/owner-and-building-info-page.tsx (hunk 8 above)
Comment (Kaemppi) [reply src/components/docx-export/pages/owner-and-building-info-page.tsx:536]: How?

File: src/components/docx-export/pages/waste-materials-page.tsx (hunk 10 above)
Comment (Kaemppi) [reply src/components/docx-export/pages/waste-materials-page.tsx:196]: How should it be?

File: src/components/docx-export/pages/reusable-materials-page.tsx (hunk 9 above)
Comment (Kaemppi) [reply src/components/docx-export/pages/reusable-materials-page.tsx:233]: ah this below.. no I dont think so

**Pull request merged**